main/streams/streams.c

```diff
@@ -561,10 +561,6 @@ PHPAPI void _php_stream_fill_read_buffer(php_stream *stream, size_t size)
 		php_stream_bucket_brigade brig_in = { NULL, NULL }, brig_out = { NULL, NULL };
 		php_stream_bucket_brigade *brig_inp = &brig_in, *brig_outp = &brig_out, *brig_swap;
 
-		/* Invalidate the existing cache, otherwise reads can fail, see note in
-		   main/streams/filter.c::_php_stream_filter_append */
-		stream->writepos = stream->readpos = 0;
```

Comment:
This was added in https://github.com/php/php-src/commit/17cfd5f24e2d73f74472a1f27e4e69ee8bea3f12 for bug https://bugs.php.net/bug.php?id=46147 . From my understanding, throwing away buffer at this point is useless, since we didn't add any new filters and buffer content should still be good.
Described issue is covered by this test here: https://github.com/php/php-src/blob/49a4e695845bf55e059e7f88e54b1111fe284223/ext/standard/tests/file/bug27619.phpt